index.bs

```diff
@@ -1693,6 +1790,140 @@ The [=remote end steps=] with |command parameters| are:
 
 </div>
 
+#### The browsingContext.navigate Command ####  {#command-browsingContext-navigate}
+
+The <dfn export for=commands>browsingContext.navigate</dfn> command navigates a
+browsing context to the given URL.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      BrowsingContextNavigateCommand = {
+        method: "browsingContext.navigate",
+        params: BrowsingContextNavigateParameters
+      }
+
+      BrowsingContextNavigateParameters = {
+        context: BrowsingContext,
+        url: text,
+        ?wait: ReadinessState,
+      }
+
+       ReadinessState = "none" / "interactive" / "complete"
+      </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+        BrowsingContextNavigateResult = {
+            navigation: Navigation / null,
+            url: text,
+        }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for browsingContext.navigate">
+The [=remote end steps=] with |command parameters| are:
+
+  1. Let |context id| be the value of the <code>context</code> field of
+     |command parameters|.
+
+  1. Let |context| be the result of [=trying=] to [=get a browsing context=]
+     with |context id|.
+
+  1. Assert: |context| is not null.
+
+  1. Let |wait condition| be the value of the <code>wait</code> field of |command
+     parameters| if present, or "<code>none</code>" otherwise.
+
+  1. Let |url| be the value of the <code>url</code> field of |command
+     parameters|.
+
+  1. Let |document| be |context|'s [=active document=].
+
+  1. Let |base| be |document|'s [=base URL=].
+
+  1. Let |url record| be the result of applying the [=URL parser=] to |url|,
+     with [=base URL=] |base|.
+
+  1. If |url record| is failure, return [=error=] with [=error code=] [=invalid
+     argument=].
+
+  1. Let |request| be a new [=/request=] whose URL is |url record|.
+
+  1. Append |context id| to the [=set of contexts with an ongoing navigate=].
+
+  1. Let |navigate status| be the result of [=navigate|navigating=] |context| with resource
+     |request|, and using |context| as the [=source browsing context=].
+
+  1. Remove |context id| from the [=set of contexts with an ongoing navigate=].
+
+  1. Let |navigation id| be |navigate status|'s id
+
+  1. If |navigate status|'s status is "<code>complete</code>" |navigation id| is
+     null:
+
+    1. Let |body| be a [=map=] matching the
+       <code>BrowsingContextNavigateResult</code> production, with the
+       <code>navigation</code> field set to null, and the <code>url</code> field
+       set to the result of the [=URL serializer=] given |navigate status|'s url.
+
+    1. Return [=success=] with data |body|, and then run the following steps [=in
+       parallel=]:
+
+       1. Run the [=WebDriver-BiDi fragment navigated=] steps given |context|
+          and |navigate status|
+
+     Note: this is the case if the navigation only caused the fragment to
```

Comment:
The navigation ID is still `null` in this case, so even though the fragmentNavigated event is emitted after the command response, there's still no way to associate the two beyond the fact that they happen to arrive side-by-side. Could we synthesize a navigation ID to expose to the client when the ID included in the navigation status is null?

Reply:
I've pushed a change on the HTML side to create a navigation id for fragment navigations.